tests/test_needs_external_needs_build.py

```diff
@@ -12,7 +12,7 @@ def test_doc_build_html(app, status, warning):
     src_dir = "doc_test/doc_needs_external_needs"
     out_dir = Path(app.outdir)
     output = subprocess.run(["sphinx-build", "-M", "html", src_dir, out_dir], capture_output=True)
-    assert not output.stderr
+    assert not output.stderr, f"Should not have stderr: {output.stderr}"
```

Comment:
What does this part do: ``f"Should not have stderr: {output.stderr}"``?

Gets ``Should not have stderr:`` written by Sphinx?

Or is it a comment? Then use ``#``.

Reply:
where is is something in the stderr, assertion fails. If so it displays the given assertion error message with the content of the stderr.
This helped me to understand problem in unit test.
I forgot to remove it when pushed. I find it makes sense to display the stderr if we are expecting to not have any.
If you want I remove this.

Reply:
Ohh cool, learned something new.
Didn't know that this gets used as output for a failed assertion.

So no, it can stay :)